frontend/src/components/MapComponent.jsx

```diff
@@ -0,0 +1,146 @@
+import PopoverComponent from "./PopOverComponent";
+import DrawComponent from "./DrawComponent";
+import SearchComponent from "./SearchComponent";
+
+const MapComponent = () => {
+  const [map, setMap] = useState(null);
+  const [vectorLayer, setVectorLayer] = useState(null);
+  const [openModal, setOpenModal] = useState(false);
+  const [selectedPlaceProperties, setSelectedPlaceProperties] = useState(null);
+  const [drawing, setDrawing] = useState(false);
+  const anchorElRef = useRef(null);
+
+  useEffect(() => {
+    useGeographic();
+    const mapInstance = new Map({
+      target: "map",
+      layers: [new TileLayer({ source: new OSM() })],
+      view: new View({
+        center: [31, 35],
+        zoom: 7,
+      }),
+    });
+    setMap(mapInstance);
+    const vectorLayer = new VectorLayer({
+      source: new VectorSource(),
+    });
+    mapInstance.addLayer(vectorLayer);
+    setVectorLayer(vectorLayer);
+
+    return () => {
+      mapInstance.setTarget(null);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!vectorLayer) return;
+    fetch('/coordinates.json')
+      .then((response) => response.json())
+      .then((data) => {
+        return new GeoJSON().readFeatures(data);
+      })
+      .then((features) => {
+        vectorLayer.getSource().addFeatures(features);
+      })
+      .catch((error) => {
+        console.error('Error loading GeoJSON:', error);
+      });
+  }, [vectorLayer]);
+
+  useEffect(() => {
+    if (!vectorLayer) return;
+    fetch('/sibiruni.json')
+      .then((response) => response.json())
+      .then((data) => {
+        return new GeoJSON().readFeatures(data);
+      })
+      .then((features) => {
+        vectorLayer.getSource().addFeatures(features);
+      })
+      .catch((error) => {
+        console.error('Error loading GeoJSON:', error);
+      });
+  }, [vectorLayer, map]);
+
+  useEffect(() => {
+    if (!map) return;
+    map.on("click", function (event) {
+      map.forEachFeatureAtPixel(event.pixel, (feature) => {
+        const properties = feature.getProperties();
+        setSelectedPlaceProperties(properties);
+        setOpenModal(true);
+        // const coordinate = feature.getGeometry().getCoordinates();
+        anchorElRef.current.style.left = event.pixel[0] + "px";
+        anchorElRef.current.style.top = event.pixel[1] + "px";
+        anchorElRef.current.style.display = "block";
+      }, {
+        layerFilter: (layer) => layer.getProperties().name && layer.getProperties().name !== 'drawLayer'
+      });
+    });
+
+    return () => {
+      map.un("click");
+    };
+  }, [map]);
+
+  const handleCloseModal = () => {
+    setOpenModal(false);
+    anchorElRef.current.style.display = "none";
+  };
+
+  const handleDelete = () => {
+    const layerToRemove = map.getLayers().getArray().find(layer => layer.getProperties().name === 'drawLayer')
+
+    if (layerToRemove) map.removeLayer(layerToRemove)
+  }
+
+  return (
```

Comment:
Given your use of material-ui, it's preferable to employ controlled components like Box, rather than relying solely on the vanilla div element